packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -888,8 +888,8 @@ DataGridPremiumRaw.propTypes = {
    */
   scrollbarSize: PropTypes.number,
   /**
-   * Set the area in `px` at the bottom of the grid viewport where onRowsScrollEnd is called.
-   * @default 80
+   * Set the percentage of the element that needs to be visible in order to invoke the rowsScrollEnd callback. A threshold of 1.0 means that when 100% of the target needs to be visible.
```

Comment:
This is a potential breaking change.

Reply:
Yes, it is also less flexible compared to the pixel value, because it doesn't allow to start fetching the next rows earlier - e.g. when 3rd row from the bottom is visible.
How do you feel about attaching a DOM node to the bottom of the content and using it for intersection? That should allow avoiding breaking changes

Reply:
I've updated it - deprecated the existing prop and added a new one. Upon further research, I couldn't replicate the behavior using the new `InterectionObserver` API.

Reply:
We can postpone it to v7. The problem is that the threshold has a different value that is not measured in pixels.

Reply:
I was thinking about this briefly, would it make sense to add an element that would always stick to the bottom of the grid and observe it with an intersection observer instead of observing the last row?
It will be possible to change this element's height and keep the same `scrollEndThreshold` API.
What do you think?

Reply:
One of the first implementations used this approach but then the feedback was to observe the last loaded row instead 😅.
This would make sense tho if we want to be backward compatible but will have to change once we do v7. At this point, we might want to park the PR for when we start the alfa and just introduce the new approach

Reply:
Yeah, backward compatibility is one of the aspects of this change. We can put it on hold until v7.

Another aspect is flexibility - in some cases, you might want to trigger row fetching way before the last row is visible. This is possible with the current `scrollEndThreshold` API, but with the new API the earliest you can fetch the next rows is when the last row enters the grid viewport.